biogeochem/EDLoggingMortalityMod.F90

```diff
@@ -296,7 +327,7 @@ subroutine LoggingMortality_frac( pft_i, dbh, canopy_layer, lmort_direct, &
             if (cur_harvest_tag == 0) then
```

Comment:
@glemieux In my CTSM testing (Derecho Intel) with `fates_harvest_mode` 1 ("fates logging via fates logging event codes") and 2 ("fates harvest driven by CLM landuse timeseries data"), I'm getting an error here that says `cur_harvest_tag` "is being used... without being defined." It is obviously _defined_, at line 240, so I think it means uninitialized.

Reply:
Just a quick update on this: I haven't tested this with https://github.com/ESCOMP/CTSM/pull/2507, but I was able to get a 1 year `fates_harvest_mode = 1` case to run without hitting this using the latest commits here and with https://github.com/E3SM-Project/E3SM/pull/6353.

Reply:
I'm still seeing it, unfortunately. 

It looks like it's possible if
- `.not. currentSite%transition_landuse_from_off_to_on`
- and `site_secondaryland_first_exceeding_min` false
- and `logging_time` true
- and `prt_params%woody(pft_i)` true

and either

- `hlm_use_lu_harvest` false

or:

- `hlm_use_lu_harvest` true
- and not `hlm_harvest_units == hlm_harvest_area_fraction`
- and not `hlm_harvest_units == hlm_harvest_carbon`

Of the either/or, my setup has the former: `hlm_use_lu_harvest` (in CTSM, `do_harvest`) is false. So it seems like we need some sort of guardrail, depending on (respectively) whether that's a valid setup or not:
- In the namelist build check: Do not allow the HLM equivalent of `hlm_use_lu_harvest` to be false if... what? `fluh_timeseries` is pointing to a file?
- In `LoggingMortality_frac()`: Immediately return (or don't get called in the first place) if the HLM equivalent of `hlm_use_lu_harvest` is false?

Reply:
@sshu88 could you help with this debugging of the `cur_harvest_tag1` variable?

Reply:
In CTSM, it looks like `hlm_harvest_units` is set by looking at the units field of the harvest rate variable on the `flanduse_timeseries` file and/or (?) the `fluh2_timeseries` file. I'm not sure if there's a way to check those units in the build namelist step. If not, we'd want to either (a) add a check as early as possible in the actual model code so that it can fail gracefully with a helpful error message, or (b) add handling to ignore harvest rates given on `flanduse_timeseries`/`fluh_timeseries` if `do_harvest` is off.

Reply:
Hi @samsrabin the prior behavior was to use the units field of the `flanduse_timeseries` file to set `flanduse_timeseries`. This should still be the case only when `fates_harvest_mode = 2`, which we've retained for backwards compatibility.  The new behavior is for `fates_harvest_mode = 3` and `fates_harvest_mode = 4`, which should be setting the `hlm_harvest_units` directly: if 3 then area-based, if 4 then mass-based.

Reply:
Okay, yes, I do see `fates_harvest_mode` 3 and 4 setting the units directly now. That explains why those aren't failing for me with `do_harvest` false. The problem, I think, is how to handle this situation:
- `do_harvest` false 
- AND [`flanduse_timeseries` and/or (?) `fluh_timeseries`] specified with harvest variable included
- AND `fates_harvest_mode` 1 or 2.

Do we want that to be a valid combination?

Reply:
Well, I guess the same question applies with `do_harvest` false and `fates_harvest_mode` 3 or 4. Any non-zero `fates_harvest_mode` feels contradictory with `do_harvest` false, so I feel like the namelist build should throw an error in such a situation.

Reply:
@samsrabin I guess that, on some level, the idea of transient land use doesn't really make sense without harvest allowed at least in principle, since only harvest can transition land from primary to secondary categories. So perhaps the simplest solution is to say that if `use_fates_luh == .true.`, then `fates_harvest_mode` must be greater than or equal to 2, and `do_harvest` must be true?

Reply:
I think that mostly makes sense, @ckoven. But when is `fates_harvest_mode` allowed to be 1?

Reply:
@samsrabin `fates_harvest_mode == 1` is only really appropriate for site-scale simulations (or maybe very highly idealized large-scale ones). But not with any of the global historical or scenario-based land use workflows.

Reply:
> @sshu88 could you help with this debugging of the `cur_harvest_tag1` variable?

Hi @ckoven @samsrabin @glemieux . I tested the global scale run but did not test site-level logging simulation. I put initialization code (https://github.com/NGEET/fates/commit/09aed9fa84a1d3032583cdd6372b763b35ba5928) and the model passed test.

Reply:
@ckoven So are these the settings we want to fail in the namelist build? (In FATES runs only, of course.)
1. `do_harvest` false AND `use_fates_luh` true — Can't have land use without harvest
2. `do_harvest` false AND `fates_harvest_mode` specified as something other than 0 — User has specified a harvest mode but not enabled harvest (could instead warn and set `do_harvest` true)
3. `do_harvest` true AND `use_fates_luh` false — Can't have harvest without land use
4. `do_harvest` true AND `fates_harvest_mode` specified as 0 — Can't have harvest without harvest
5. `do_harvest` true AND `fates_harvest_mode` specified but mismatched with units on `fluh_timeseries` — We don't want the user's intention to be ignored by the code parsing `fluh_timeseries`
6. `do_harvest` true AND `fates_harvest_mode` 1 AND `fluh_timeseries` specified

Sorry to be such a pill about this, but I know deep in my bones I will get myself into trouble without these kinds of guardrails!

Reply:
Also:
7. `do_harvest` true in non-transient run

Anything else special for non-transient runs?

Reply:
No worries, there are good questions! 

I think these are correct, although I have followup on the following:
2. Do we have users who just want to use the logging codes and not landuse harvest?
1/3. Are there scientifically relevant simulations to do with landuse but no harvesting?

Also, I think we should move away from using `do_harvest` at all for fates.  I.E. I think `do_harvest` should only be for HLM-specific harvest calls like `CNHarvest`.   I'm working on an update that disentangles this on the HLM side.  I've also started a PR to draft a page in the user's guide explaining the various `fates_harvest_modes` in more detail to help drive this distinction: https://github.com/NGEET/fates-users-guide/pull/63

Reply:
Are you suggesting we fully deprecate `do_harvest` as a user-facing setting? If so, I'm totally on board. It seems redundant with `fates_harvest_mode`.

I'll let Charlie comment on your follow-up questions.

Reply:
> Are you suggesting we fully deprecate `do_harvest` as a user-facing setting? If so, I'm totally on board. It seems redundant with `fates_harvest_mode`.

Yep, exactly.  Now that fates harvesting options are moving beyond the HLM harvest modes I think it makes it easier to manage.  Started up a PR to your ctsm PR here: https://github.com/samsrabin/CTSM/pull/2

Reply:
@samsrabin I think we should allow harvest without the full land use change drivers, I could imagine someone wanting to do that experiment. So I would allow your number 3 (or the variant of it in which we remove `do_harvest` entirely: to allow harvest mode to be any value of 1-4 when `use_fates_luh == false`. Otherwise I agree with al the above.